Source/driver/Castro_advance.cpp

```diff
@@ -112,7 +112,8 @@ Castro::advance (Real time,
     } else {
       for (int iter = 0; iter < MOL_STAGES; ++iter) {
 	mol_iteration = iter;
-	dt_new = do_advance(time, dt, amr_iteration, amr_ncycle);
+        // TODO: should be time or time + c_mol[iter]*dt?
```

Comment:
Is this question resolved?

Reply:
yes.  I think it is correct now -- with the trick of making the old and new sources consistent.